src/packages/solid/components/Badge/Badge.tsx

```diff
@@ -14,39 +14,35 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-import { type Component } from 'solid-js';
-import { Text, type TextProps } from '@lightningjs/solid';
-import { withPadding } from '../../utils/index.js';
-import type { UIComponentProps } from '../../types/interfaces.js';
+import { createMemo, type Component } from 'solid-js';
+import { Text } from '@lightningjs/solid';
+import { withPadding } from '@lightningjs/solid-primitives';
 import styles from './Badge.styles.js';
+import type { BadgeProps } from './Badge.types.js';
 withPadding; // Preserve the import.
 
-interface BadgeProps extends UIComponentProps {
-  /**
-   * Badge text
-   */
-  title: TextProps['children'];
+const getTone = (props: BadgeProps) => props.tone ?? styles.tone;
+const getTitle = (title: string | string[]) => title ?? '';
+const getPadding = (props: BadgeProps, tone: string) =>
+  props.padding ?? styles.Container?.tones[tone]?.padding ?? styles.Container.base.padding;
 
-  padding?: number[];
-}
+const BadgeContainer: Component<BadgeProps> = props => {
+  const tone = createMemo(() => getTone(props));
+  const padding = createMemo(() => getPadding(props, tone()));
 
-interface BadgeContainerProps extends UIComponentProps {
-  padding?: number[];
-}
-
-const BadgeContainer: Component<BadgeContainerProps> = props => {
   return (
     <node
-      use:withPadding={
-        props.padding ??
-        styles.Container?.tones[props.tone ?? styles.tone]?.padding ??
-        styles.Container.base.padding
-      }
+      use:withPadding={padding()}
       {...props}
+      borderColor={props.strokeColor} // TODO clew uses strokeColor, but we currently don't account for nested properties (border.color)
+      borderWidth={props.strokeWidth} // TODO clew uses strokeWidth, but we currently don't account for nested properties (border.width)
```

Comment:
Since these won't actually do anything, we should remove them from here and then add these comments to the style file and comment them out temporarily. I think it's best to keep the markup as clean as possible